integration_test/tests/power_bi_tests.rs

```diff
@@ -368,4 +370,165 @@ mod integration {
+            );
+
+            // SQL_DRIVER_ODBC_VER and SQL_DRIVER_NAME are not available through odbc_sys
+            /*
+            SQLGetInfoW(SQL_DRIVER_ODBC_VER)
+            SQLGetInfoW(SQL_DRIVER_NAME)
+            */
+
+            let mut query: Vec<u16> = "select * from example".encode_utf16().collect();
+            query.push(0);
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLExecDirectW(stmt as HStmt, query.as_ptr(), SQL_NTS as i32),
+                "{}",
+                get_sql_diagnostics(HandleType::Stmt, stmt as Handle)
+            );
+
+            // SQLGetFunctions is not available through odbc_sys
+            /*
+            SQLGetFunctions(SQL_API_SQLFETCHSCROLL)
+            */
+
+            let str_len_ptr = &mut 0;
+            let output_buffer = &mut [0u16; (BUFFER_LENGTH as usize - 1)] as *mut _;
+
+            test_get_info!(
+                conn_handle,
+                InfoType::GetDataExtensions,
+                2,
+                DataType::USmallInt
+            );
+
+            let column_count_ptr = &mut 0;
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLNumResultCols(stmt as HStmt, column_count_ptr)
+            );
+            assert_eq!(2, *column_count_ptr);
+
+            let numeric_attribute_ptr = &mut 0;
+            const FIELD_IDS: [Desc; 7] = [
+                Desc::ConciseType,
+                Desc::Unsigned,
+                Desc::Name,
+                Desc::Nullable,
+                Desc::TypeName,
+                Desc::Length,
+                Desc::Scale,
+            ];
+            for col_num in 0..*column_count_ptr {
+                FIELD_IDS.iter().for_each(|field_type| {
+                    assert_eq!(
+                        SqlReturn::SUCCESS,
+                        SQLColAttributeW(
+                            stmt as HStmt,
+                            (col_num + 1) as u16,
+                            *field_type,
+                            output_buffer as Pointer,
+                            14,
+                            str_len_ptr,
+                            numeric_attribute_ptr,
+                        ),
+                        "{}",
+                        get_sql_diagnostics(HandleType::Stmt, stmt as Handle)
+                    );
+                });
+            }
+
+            let mut successful_fetch_count = 0;
+            loop {
+                let result = SQLFetch(stmt as HStmt);
+                assert!(
+                    result == SqlReturn::SUCCESS || result == SqlReturn::NO_DATA,
+                    "{}",
+                    get_sql_diagnostics(HandleType::Stmt, stmt as Handle)
+                );
+                match result {
+                    SqlReturn::SUCCESS => {
+                        successful_fetch_count += 1;
+                        assert_eq!(
+                            SqlReturn::SUCCESS,
+                            SQLGetData(
```

Comment:
[nit] Technically, you could have a loop calling SQLGetData and using the info returned by the SQLColAttribute calls to pick the target type to ask for. This way you would not be dependent on you knowing that there are exactly 2 columns in the tables and that the first one is a long, the second a string

Reply:
I thought about this too, and wasn't sure whether that would be more or less clear. Thinking about it a bit more, I think it is worth doing so I can make this change!